source/normalize-arguments.ts

```diff
@@ -62,27 +67,28 @@ export const preNormalizeArguments = (options: any, defaults?: any) => {
 		methods: new Set(),
 		statusCodes: new Set(),
 		errorCodes: new Set(),
-		maxRetryAfter: undefined
+		maxRetryAfter: 0
 	};
 
 	if (is.nonEmptyObject(defaults) && retry !== false) {
-		options.retry = {...defaults.retry};
+		options.retry = {...(defaults.retry as RetryOption)};
 	}
 
 	if (retry !== false) {
 		if (is.number(retry)) {
 			options.retry.retries = retry;
 		} else {
-			options.retry = {...options.retry, ...retry};
+			// eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+			options.retry = {...retry, ...options.retry} as NormalizedRetryOptions;
```

Comment:
This is incorrect. It overwrites your `retry` config with the defaults,
thus causing `timeout -> retries on timeout` to fail:
https://travis-ci.org/sindresorhus/got/jobs/545268544#L832
```suggestion
			options.retry = {...options.retry, ...retry} as NormalizedRetryOptions;
```